### DISCUSSION TITLE
The dummy's Ideas

### DISCUSSION THREAD

**Discussion by Jakey-Jakester** [domtronn/nuzlocke.app#24] [Ideas]
Howdy, I know very little about software development, and am asking a lot for a one man dev team, but here are a few ideas...

NPC locations
Item Locations
Evil Team Fights to nuzlocke routing
Difficult Battle Marker
Ability to show level caps but hide the actual Pokémon of a rival/gym battle.
Static Encounters + gifts
Ability to view Pokémon properties Per ROM (evolution requirements, Moves learnt, Move tutor stuff, Egg moves, etc.)
Ability to edit Pokémon properties per rom
Ability to create ROM templets (Modifiable routes, battles, Pokémon per route, Pokémon move pools and evolution requirements, What HM's are needed for where, etc.)
Ability to create rulesets (and maybe have them effect the nuzlocke. For example allowing Gift Pokémon would add certain static encounters, Monotype runs would change the Pokémon suggested per route, etc.)
Suggested rules to add (Shiny clause, randomized starter, duplicate clause, 
Ability to share rulesets and ROM templets easily (Maybe as a file, or as 
Ability to give Pokémon "lives" or add a place to "revive" Pokémon. (I saw people on reddit sometimes play where their Pokémon  have multiple lives) (Also In my renegade Platinum Nuzlocke Run, I am reviving 6 Pokémon when I reach Pal Park, and it would be super sick to be able to have this shown in the tracker)
Ability to mark a battle as particularly difficult
Ability to quick add a Pokémon (For example a shiny, and maybe this button could even be a toggle that checks for if a shiny clause has been chosen)
item tracker (can be influenced by what's available on route)
Ability to mark in what battle a Pokémon died (Specific Trainer battle, adventuring, grinding)
HM tracker
The ability to, after a run, show where each Pokémon was caught, and where they died, sort of creating a little story of the nuzlocke!
The ability to save our old nuzlockes to an account or as a file
Maybe the ability to upload screenshots of a Pokémon, and have the app scan it for certain text and then put in the data about a given Pokémon for the player, but this sounds very hard
The ability to support you some how (maybe through a 1 time In App Purchase, or through (non-intrusive) ads or through a patreon or whatever else)

**Comment by Jakey-Jakester** [comment]
Also maybe have the icons on the app update to show gender, shiny, or give the user the ability to upload their own art. It would also be sick if you could show the user data across multiple different nuzlockes, like their favorite types, moves, amount of deaths, etc. Also Also if you add the item tracker and ability to change Pokémon data (move sets mostly, but also stats and evolution requirements) per rom it would be cool to suggest what moves a Pokémon should learn for an upcoming gym or rival battle, or just to round out the team and cover more typing.

**Comment by domtronn** [comment]
Hey @Jakey-Jakester! I really appreciate all the ideas, this is fantastic... you're right though, it is a lot to digest but I'll do my best!

I like a lot of these and it's great to hear a fresh set of ideas.
One thing I'd want to caveat is that most of the data for this app is [manually curated](https://github.com/domtronn/dc-nuzlocke-data) and the more data I add the slower/more slugish the app could become _(or at least the harder it becomes to optimise!)_ and when this information exists readily in things like Serebii or Bulbapedia I don't want to duplicate the content they've already created beyond what's really useful to a nuzlocke run.

I've tried to reply to each point but I think I've been able to summarise the features as follows

1. **Tell a story** - Expand on the ability to use the app to tell the story of your Nuzlocke. Things like summary pages (catches, deaths, type distribution) as well as Deaths to trainers/gyms and particularly hard battles

2. **Rulesets** - Allow users to select / customise their own ruleset and have that codify certain behaviour. For example dupes clause warns when encountering the same thing twice, shiny clause lets you mark and add shiny pokemon, monotype run filters the route to just that type

3. **Custom locations** - Allow users to add a custom location, either because the data is wrong or for times like shiny clauses.

---

> NPC locations

So each "boss battle" includes the bosses location so you can be prepared. I think other NPCs such as move relearner etc are probably best left in game Wikis for fear of duplicating content.

> Item Locations
> Item tracker (can be influenced by what's available on route)

Similar to the above point, but I appreciate held items are way more important for Nuzlockes so I like the idea of highlighting these next to a route so you don't miss them! 

> Evil Team Fights to nuzlocke routing

I'm not sure what you mean by this?

> Ability to show level caps but hide the actual Pokémon of a rival/gym battle.

Since the boss teams are in accordions they're kind of hidden already if you don't open them, but you're right that on desktop you get a little "preview". Would be a nice feature for a settings page 👍 

> Static Encounters + gifts

So I _should_ have included a location for you to enter each static encounter. If I've missed any in the RomHacks, let me know and I can add them in!

> Ability to view Pokémon properties Per ROM (evolution requirements, Moves learnt, Move tutor stuff, Egg moves, etc.)

I get that this information can be hard to find for rom hacks, so would be nice to centralise / standardise somehow but currently since RomHacks are a beta feature I'd want to focus on the core / main series experience. At which point I again don't want to duplicate data that already exists in other sources.

> Ability to edit Pokémon properties per rom
> Ability to create ROM templets (Modifiable routes, battles, Pokémon per route, Pokémon move pools and evolution requirements, 

This is already possible using the [rom hack patches](https://github.com/domtronn/dc-nuzlocke-data/blob/main/patches/renplat.txt). This lets you change types, moves, items, and pokemon stats. Combine this with the [routes](https://github.com/domtronn/dc-nuzlocke-data/blob/main/routes/renplat.txt) and you can make a completely custom game! Albeit currently I need to merge it in, which is purposeful so others can enjoy them too.

> ~What HM's are needed for where, etc.)~
> ~HM tracker~

> Ability to create rulesets (and maybe have them effect the nuzlocke. For example allowing Gift Pokémon would add certain static encounters, Monotype runs would change the Pokémon suggested per route, etc.)
> Suggested rules to add (Shiny clause, randomized starter, duplicate clause,
> Ability to share rulesets and ROM templets easily (Maybe as a file, or as

Rules is a feature on the backlog - I think there's a lot more I can do to help users "tell a story" of their nuzlocke, and sharing their rulesets is clearly one of them. To begin with I just assumed people would use the app with whatever ruleset they wanted to play with. However, things like dupes clause, shiny clause would be nice to codify! And I really like the Monotype run idea, I think that would be great to filter the routes by type availability.

> Ability to give Pokémon "lives" or add a place to "revive" Pokémon. (I saw people on reddit sometimes play where their Pokémon have multiple lives) (Also In my renegade Platinum Nuzlocke Run, I am reviving 6 Pokémon when I reach Pal Park, and it would be super sick to be able to have this shown in the tracker)

Hmmm, I'd never heard of this rule but an interesting feature, fits into the rulesets feature nicely. But for now, you can always revive pokemon.

> Ability to mark a battle as particularly difficult

Again, I think this fits into the "tell a story" feature as a whole. And I like this idea! Clearly by the end of a nuzlocke you want to share what you found hard and what happened so I like this as an idea. I'm currently working on a "Progress" feature that lets you mark off battles as you go - this could prompt you for some kind of difficulty rating.

> Ability to quick add a Pokémon (For example a shiny, and maybe this button could even be a toggle that checks for if a shiny clause has been chosen)

The "Custom Locations" feature should cover this. If people are playing with shiny clause they could add a Custom Location for that shiny.

> Ability to mark in what battle a Pokémon died (Specific Trainer battle, adventuring, grinding)
> The ability to, after a run, show where each Pokémon was caught, and where they died, sort of creating a little story of the nuzlocke!

I thought about this before, but I think you're right that this helps "tell a story" again. I already have the capture data which you can see on the Box page, the death data would need to be added but could have a little prompt when someone selects "Dead" from the dropdown.

> The ability to save our old nuzlockes to an account or as a file
> Maybe the ability to upload screenshots of a Pokémon, and have the app scan it for certain text and then put in the data about a given Pokémon for the player, but this sounds very hard

Haha, love it! But yes, certainly very hard... Definitely one to consider if I ever run out of other ideas ;) 
But I have tried to keep the data entry as automated / minimal as possible so keep them up to date isn't a chore

> Also maybe have the icons on the app update to show gender, shiny, or give the user the ability to upload their own art. It would also be sick if you could show the user data across multiple different nuzlockes, like their favorite types, moves, amount of deaths, etc. 

I **love** this idea. I've got two features in the works, a team builder & a graveyard which I think will go a long way to help this... But having it across _all_ your save files would be awesome. Definitely going to do this as part of the **summary** feature. _(sneak peek below)_ 
![2021-12-12 11 05 15](https://user-images.githubusercontent.com/5881414/145709828-ebd12aaf-8360-4d70-8725-978edc5c1599.gif)
![2021-12-12 11 08 28](https://user-images.githubusercontent.com/5881414/145709906-de657e2d-7f84-4571-b808-756ea65f7672.gif)

> Also Also if you add the item tracker and ability to change Pokémon data (move sets mostly, but also stats and evolution requirements) per rom it would be cool to suggest what moves a Pokémon should learn for an upcoming gym or rival battle, or just to round out the team and cover more typing.

Obviously the point of the app is to help with the theory crafting for a fight. I'm very bad at it hence why I built this app! But I also think part of the fun is coming up with the right team and moveset yourself. I don't want this to just "tell you the answer", but give you the information to help you find the right answer! I played with this idea once but felt the Compare info that shows the opponents type weaknesses and stat weaknesses was enough to help me.

---

> The ability to support you some how (maybe through a 1 time In App Purchase, or through (non-intrusive) ads or through a patreon or whatever else)

That's very kind! I plan to set up something like a Patreon or a Buy Me A Coffee eventually but for now just enjoy it and please keep providing feedback 😄

**Reply by Jakey-Jakester** [reply]
Let me say that one of the best parts of using your app is the little animations. Please keep adding them as they do NOT go unappreciated!!! Thank you so much for taking the time to read my ideas!

**Reply by Jakey-Jakester** [reply]
For the graveyard thing, I think it would be cool for you to maybe add almost like superlative awards, like which of the dead pokemon was highest level, which would have been helpful in the gym before they died, which went on the longest journey, etc etc

**Reply by Jakey-Jakester** [reply]
On the home screen of the app, It would be cool if the pokemon that appear next to the continue, new game, and load game were Pokémon cought/dead and if you hovered over them you could see info on them.
